cephalopod/utils.py

```diff
@@ -123,3 +124,12 @@ def aggregate_chart(extended_instances, extra_fields):
             }
 
     return aggregated_fields
+
+
+def login_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if 'user' not in session:
```

Comment:
Check the whitelist again here. Otherwise a session is valid forever as long as someone has the cookie.